common/scripted_effects/wc_magic_light_effects.txt

```diff
@@ -0,0 +1,1387 @@
+            }
+        }
+        change_county_control = -100
+        change_development_level = -10
+        custom_tooltip = wc_divine_storm_tt
+        every_in_de_jure_hierarchy = {
+            if = {
+                limit = {
+                    tier = tier_county
+                }
+                add_county_modifier = {
+                    modifier = wc_divine_storm_county_modifier
+                    days = wc_current_spell_duration
+                }
+                trigger_event = {
+                    on_action = wc_divine_storm_county_pulse
+                    months = 1
+                }
+            }
+            else_if = {
+                limit = {
+                    tier = tier_barony
+                }
+                title_province = {
+                    hidden_effect = {
+                        every_character_in_location = {
+                            limit = {
+                                NOT = {
+                                    has_character_modifier = wc_divine_storm_character_modifier
+                                }
+                            }
+                            apply_character_divine_storm_effect = yes
+                        }
+                    }
+                }
+            }
+        }
+    }
+}
+apply_character_divine_storm_effect = {
+    if = {
+        limit = {
+            NOT = {
+                has_character_modifier = wc_divine_storm_character_modifier
+            }
+        }
+        if = {
+            limit = {
+                OR = {
+                    culture = { has_same_culture_heritage = culture:eredruin }
+                    culture = { has_same_culture_heritage = culture:scourge }
+                    has_trait = being_void
+                    has_trait = being_demon
+                    has_trait = being_undead
+                }
+            }
+            random_list = {
+                40 = {
+                    add_or_increase_burned_effect = yes
+                }
+                40 = {
+                    if = {
+                        limit = {
+                            NOT = { has_trait = wounded_1 }
+                        }
+                        add_trait = wounded_1
+                    }
+                    else = {
+                        change_trait_rank = { trait = wounded rank = 1 max = 3 }
+                    }
+                }
+                30 = {
+                    modifier = {
+                        always = yes
+                        subtract = wc_light_magic_resistance_value
+                    }
+                    override_death_killer_effect = { death_reason = death_divine_storm killer = root }
+                }
+                30 = {
+                    add_trait = holy_wrath
+                }
+            }
+        }
+        add_character_modifier = {
+            modifier = wc_divine_storm_character_modifier
+            days = wc_current_spell_duration
```

Comment:
You aren't saving the duration before using it in on_actions. It will be applied forever.